contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6181,11 +6184,40 @@ transformSelectIntoStmt(CreateTableAsStmt *stmt)
 			}
 			else
 			{
+				AlterTableStmt *newstmt;
+				AlterTableCmd *cmd;
+				char* original_name = NULL;
+				newstmt = NULL;
+
+				if (tle->resname != NULL)
+				{
+					original_name = tle->resname;
+					tle->resname = downcase_identifier(tle->resname, strlen(tle->resname), false, false);
+				}	
+
 				current_resno += 1;
 				tle->resno = current_resno;
 				modifiedTargetList = lappend(modifiedTargetList, tle);
+
+				if (original_name != NULL && strcmp(original_name, tle->resname) && strlen(original_name) == NAMEDATALEN)
+				{
```

Comment:
`strlen(original_name) < NAMEDATALEN`
only then we save the original name.

Reply:
actually truncation makes the name < NAMEDATALEN. so we may not be able to tell if a name is truncated or not just by looking at the length.